File: 3P/3P PROYECTO/Busqueda con Arboles - BIG O/02. Codigo/big_O.m
% Leer el archivo CSV
data = dlmread("resultados.csv", ";");

% Obtener los datos de las columnas
x_data = data(:, 1);
y_data = data(:, 2);

% Crear una línea de color rosa que une los puntos
plot(x_data, y_data, 'm-', 'LineWidth', 2);

xlabel('Número de Datos');
ylabel('Tiempo de Ejecución (ms)');
title('Big O en Octave');
grid on;

% Pausa durante 5 segundos para visualizar la gráfica
pause(10);

File: 3P/3P PROYECTO/Busqueda con Arboles - BIG O/02. Codigo/big_O.R
# Cargar el paquete ggplot2
library(ggplot2)

# Leer el archivo CSV
data <- read.csv("resultados.csv", header = TRUE, sep = ";")

# Obtener los nombres de las columnas
x_name <- names(data)[1]
y_name <- names(data)[2]

# Crear la gráfica utilizando ggplot2 con aes()
plot <- ggplot(data, aes(x = !!sym(x_name), y = !!sym(y_name))) +
  geom_line(color = "purple", linewidth = 1.5) +  # Línea de color morado y más gruesa
  labs(x = x_name, y = y_name) +
  ggtitle("Big O en R")

# Mostrar la gráfica en una ventana gráfica independiente
if (Sys.info()["sysname"] == "Windows") {
  windows()
} else {
  x11()
}

# Mostrar la gráfica en la ventana gráfica independiente
print(plot)

# Pausa para que la ventana no se cierre inmediatamente
Sys.sleep(10)  # Pausa de 10 segundos (ajusta el valor según lo necesario)
